fetch.js

```diff
@@ -459,6 +469,10 @@
         xhr.setRequestHeader(name, value)
       })
 
+      if (request.signal && request.signal.addEventListener) {
+        request.signal.addEventListener('abort', xhr.abort)
```

Comment:
Here, `event` object will be passed to `xhr.abort()` as an argument, which probably doesn't have side-effects, but to avoid it let's create a separate function for the event handler that calls into `xhr.abort()`.